### PR TITLE
DDF-1408: Converted and Added UI Tests For Installation

### PR DESCRIPTION
**Pull request by AzGoalie**
Converted and added tests for the installation UI.

@ricklarsen will hero
@lessarderic 
@pklinef 
@brendan-hofmann

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/167)

### CODE REVIEW

**Comment by ricklarsen** [comment]
Reviewed 16 of 16 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/167)

**Comment by brendan-hofmann** [comment]
Reviewed 16 of 16 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

<sup>**[platform/admin/ui/Gruntfile.js, line 126 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/167#-Jyrky8Xo1wl8umJFWe3)** ([raw file](https://github.com/codice/ddf/blob/820f1b36c0625c1d4f0bbfff80fa317bfe9216fd/platform/admin/ui/Gruntfile.js#L126)):</sup>
Is this right? We don't test browsers on any other platforms?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/167)

**Comment by ricklarsen** [comment]
Successfully built and tested.

**Comment by lessarderic** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[platform/admin/ui/server-impl.js, line 134 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/167#-Jys57CDQYfDN2E2VvEb)** ([raw file](https://github.com/codice/ddf/blob/820f1b36c0625c1d4f0bbfff80fa317bfe9216fd/platform/admin/ui/server-impl.js#L134)):</sup>
Can this ever be true? Shouldn't we always use {} blocks in if/else statements?

Also, do we need the mocks inside the implementation class or could we move them to a separate file?

---

<sup>**[platform/admin/ui/src/test/js/wd/InstallationTests.js, line 23 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/167#-Jys8ujswwoJvXfdt4C5)** ([raw file](https://github.com/codice/ddf/blob/820f1b36c0625c1d4f0bbfff80fa317bfe9216fd/platform/admin/ui/src/test/js/wd/InstallationTests.js#L23)):</sup>
Can't we use id "collapseAlerts" instead of the CSS selector?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/167)

**Comment by lessarderic** [comment]
Reviewed 16 of 16 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/167)

**Comment by AzGoalie** [comment]
Review status: 14 of 16 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

<sup>**[platform/admin/ui/Gruntfile.js, line 126 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/167#-Jyrky8Xo1wl8umJFWe3)** ([raw file](https://github.com/codice/ddf/blob/820f1b36c0625c1d4f0bbfff80fa317bfe9216fd/platform/admin/ui/Gruntfile.js#L126)):</sup>
This is what we were testing in the search UI, i can add other platforms if needed. Should it be the three main ones (Windows/Mac/Linux)? 

---

<sup>**[platform/admin/ui/src/test/js/wd/InstallationTests.js, line 23 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/167#-Jys8ujswwoJvXfdt4C5)** ([raw file](https://github.com/codice/ddf/blob/820f1b36c0625c1d4f0bbfff80fa317bfe9216fd/platform/admin/ui/src/test/js/wd/InstallationTests.js#L23)):</sup>
Fixed

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/167)

**Comment by brendan-hofmann** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[platform/admin/ui/Gruntfile.js, line 126 \[r1\]](https://reviewable.io:443/reviews/codice/ddf/167#-Jyrky8Xo1wl8umJFWe3)** ([raw file](https://github.com/codice/ddf/blob/820f1b36c0625c1d4f0bbfff80fa317bfe9216fd/platform/admin/ui/Gruntfile.js#L126)):</sup>
That's what I would have expected, @pklinef please correct me if I'm wrong.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/167)

File: platform/admin/ui/package.json [hunk 1]
```diff
@@ -22,21 +22,28 @@
     "node": ">=0.10.5"
   },
   "devDependencies": {
-    "grunt": "0.4.5",
-    "grunt-cli": "0.1.13",
     "bower": "~1.4.1",
+    "chai": "^1.10.0",
```
Comment (pklinef) [inline platform/admin/ui/package.json]: Don't add versions with ^ or ~.  There is another ticket removing those.

File: platform/admin/ui/src/test/js/wd/InstallationTests.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                .waitForConditionInBrowser("document.getElementsByClassName('claim-table')[0].rows.length === 3", shared.timeout)
+                .waitForElementByCssSelector('input[value="http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier"]', shared.timeout)
+                .waitForElementByCssSelector('input[value="http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role"]', shared.timeout);
+        });
+
+        it('should have defaults for dropdown', function() {
+            return this.browser
+                .waitForElementByClassName('fa-caret-down', shared.timeout).click()
+                .waitForElementByClassName('dropdown-menu', asserters.isDisplayed, shared.timeout)
+                .takeScreenshot().saveScreenshot(shared.getPathForScreenshot('attribute-dropdown.png'))
```
Comment (pklinef) [inline platform/admin/ui/src/test/js/wd/InstallationTests.js]: Taking screenshots is slow so I would suggest removing these unless they are actually be used.

File: platform/admin/ui/src/test/js/wd/InstallationTests.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                    .waitForConditionInBrowser("document.querySelectorAll('input#content-appcb:checked').length == 1", shared.timeout)
+                    .waitForConditionInBrowser("document.querySelectorAll('input#catalog-appcb:checked').length == 1", shared.timeout)
+                    .waitForConditionInBrowser("document.querySelectorAll('input#platform-appcb:checked').length == 1", shared.timeout);
+            });
+
+            it('should display details when hovering over an app', function() {
+                return this.browser
+                    .waitForElementById('solr-appcb', shared.timeout).moveTo()
+                    .waitForElementByCssSelector('td#detailsName', shared.timeout)
+                    .takeScreenshot().saveScreenshot(shared.getPathForScreenshot('hover.png'))
```
Comment (pklinef) [inline platform/admin/ui/src/test/js/wd/InstallationTests.js]: Another screenshot.

File: platform/admin/ui/Gruntfile.js [hunk 4]
```diff
@@ -89,14 +87,48 @@ module.exports = function (grunt) {
...
+                    port: 4444,
+                    // mochaWebdriver always starts a selenium server so
+                    // starting phantomjs instance that will not be used
+                    phantomPort: 5555,
+                    usePhantom: true
                 }
             },
-            //this is where the tests would be called from
-            files: ['src/test/js/*.js']
+            sauce: {
```
Comment (pklinef) [inline platform/admin/ui/Gruntfile.js]: Did you try running these tests against Sauce Labs and verify that it worked with all three of these browsers on Windows?

**Comment by AzGoalie** [comment]
Review status: 14 of 16 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[platform/admin/ui/Gruntfile.js, line 118 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/167#-JyoL6LSd0ZQJNF45Ij9-r2-118)** ([raw file](https://github.com/codice/ddf/blob/586f70bdca8409fb2e54b647c15fc6f6b90f34c9/platform/admin/ui/Gruntfile.js#L118)):</sup>
Yes, chrome and internet explorer work, firefox has a bug.

---

<sup>**[platform/admin/ui/package.json, line 26 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/167#-JyoL6LSd0ZQJNF45IjA-r2-26)** ([raw file](https://github.com/codice/ddf/blob/586f70bdca8409fb2e54b647c15fc6f6b90f34c9/platform/admin/ui/package.json#L26)):</sup>
Done.

---

<sup>**[platform/admin/ui/src/test/js/wd/InstallationTests.js, line 87 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/167#-JyoL6LSd0ZQJNF45IjE-r2-87)** ([raw file](https://github.com/codice/ddf/blob/586f70bdca8409fb2e54b647c15fc6f6b90f34c9/platform/admin/ui/src/test/js/wd/InstallationTests.js#L87)):</sup>
Done.

---

<sup>**[platform/admin/ui/src/test/js/wd/InstallationTests.js, line 237 \[r2\]](https://reviewable.io:443/reviews/codice/ddf/167#-JyoL6LSd0ZQJNF45IjE-r2-237)** ([raw file](https://github.com/codice/ddf/blob/586f70bdca8409fb2e54b647c15fc6f6b90f34c9/platform/admin/ui/src/test/js/wd/InstallationTests.js#L237)):</sup>
Done.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/167)

**Comment by brendan-hofmann** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/167)

File: platform/admin/ui/test.js [hunk 5]
```diff
@@ -27,7 +27,17 @@ app.use(express.static(__dirname + '/src/main/webapp'));
...
+app.all('/jolokia/read/org.codice.ddf.admin.application.service.ApplicationService:service=application-service/InstallationProfiles/', server.mockInstallationProfiles);
+app.all('/jolokia/exec/org.codice.ddf.ui.admin.api.ConfigurationAdmin:service=ui,version=2.3.0/getService/*', server.mockGetService);
+app.all('/jolokia/exec/org.codice.ddf.ui.admin.api.ConfigurationAdmin:service=ui,version=2.3.0/getClaimsConfiguration/*', server.mockClaims);
+app.all('/jolokia/exec/org.codice.ddf.ui.admin.api.ConfigurationAdmin:service=ui,version=2.3.0/add', server.mockRequest);
+app.all('/jolokia/exec/org.codice.ddf.ui.admin.api.ConfigurationAdmin:service=ui,version=2.3.0/listServices', server.mockRequest);
+app.all('/jolokia/read/org.codice.ddf.admin.application.service.ApplicationService:service=application-service/ApplicationTree/', server.mockAppTree);
+app.all('/jolokia/exec/org.codice.ddf.admin.application.service.ApplicationService:service=application-service/startApplication/*', server.mockStart);
+//app.all('/jolokia/*', server.requestProxy);
```
Comment (pklinef) [inline platform/admin/ui/test.js]: Remove these if you don't need them.

**Comment by brendan-hofmann** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/167)

**Pull request merged**